src/components/AddNewCharacterForm/AddNewCharacterForm.js

```diff
@@ -0,0 +1,233 @@
+import { Button, Container, Grid, makeStyles } from "@material-ui/core"
+import { useForm } from "react-hook-form"
+import { useDispatch } from "react-redux"
+import { nanoid } from "@reduxjs/toolkit"
+
+import { addCharacter } from "@/reducers/characterReducer"
+
+import ControlledInputField from "../ControlledInputField"
+import ControlledSelectInputField from "../ControlledSelectInputField/ControlledSelectInputField"
+
+const useStyles = makeStyles((theme) => ({
+  title: {
+    borderBottom: "1px solid grey",
+    padding: theme.spacing(1),
+    fontWeight: theme.typography.fontWeightBold,
+  },
+
+  body: {
+    padding: "15px",
+    display: "flex",
+    flexDirection: "column",
+    gap: theme.spacing(2),
+    width: "30vw",
+  },
+
+  buttonBox: {
+    display: "flex",
+    gap: theme.spacing(2),
+    position: "absolute",
+    bottom: 0,
+    padding: theme.spacing(2),
+    "& > *": {
+      color: theme.palette.common.white,
+    },
+    "& > :first-child": {
+      backgroundColor: theme.palette.button.main,
+    },
+    "& > :last-child": {
+      backgroundColor: theme.palette.grey[400],
+    },
+  },
+}))
+
+const AddNewCharacterForm = ({ setOpen }) => {
+  const classes = useStyles()
+  const { control, handleSubmit } = useForm({
+    defaultValues: {
+      name: "",
+      eyeColor: "",
+      height: "",
+      gender: "",
+      birthYear: "",
+      homeworld: "",
+      species: "",
+      numberOfFilms: "",
+    },
+    reValidateMode: "onSubmit",
+  })
+
+  const dispatch = useDispatch()
+
+  const onSubmit = (data) => {
+    if (data.name && data.eyeColor) {
```

Comment:
We like to favor early returns in conditionals because it makes for cleaner code with smaller code blocks, you can change this to 
```
if (!data.name || !data.eyeColor) return;
```
But we will be changing this to be done in validation anyway, but keep this in mind